arch/x86/core/ia32/cache.c

```diff
@@ -47,66 +47,18 @@ extern size_t z_cache_line_size_get(void);
  * @return N/A
  */
 
-_sys_cache_flush_sig(_cache_flush_clflush)
+void arch_cache_flush(void *start_addr, size_t size)
 {
-	int end;
+	u32_t end;
 
-	size = ROUND_UP(size, sys_cache_line_size);
-	end = virt + size;
+//	size = ROUND_UP(size, z_cache_line_size_get());
+	end = (u32_t)start_addr + size;
 
-	for (; virt < end; virt += sys_cache_line_size) {
-		__asm__ volatile("clflush %0;\n\t" :  : "m"(virt));
+	for (; (u32_t)start_addr < end; (u32_t)start_addr += z_cache_line_size_get()) {
+		__asm__ volatile("clflush %0;\n\t" :  : "m"((u32_t)start_addr));
 	}
 
 	__asm__ volatile("mfence;\n\t");
 }
 
 #endif /* CONFIG_CLFLUSH_INSTRUCTION_SUPPORTED || CLFLUSH_DETECT */
-
-#if defined(CONFIG_CLFLUSH_DETECT) || defined(CONFIG_CACHE_LINE_SIZE_DETECT)
-
-#include <init.h>
-
-#if defined(CONFIG_CLFLUSH_DETECT)
-_sys_cache_flush_t *sys_cache_flush;
-static void init_cache_flush(void)
-{
-	if (z_is_clflush_available()) {
-		sys_cache_flush = _cache_flush_clflush;
-	} else {
-		sys_cache_flush = z_cache_flush_wbinvd;
-	}
-}
-#else
-#define init_cache_flush() do { } while (false)
-
-#if defined(CONFIG_CLFLUSH_INSTRUCTION_SUPPORTED)
-FUNC_ALIAS(_cache_flush_clflush, sys_cache_flush, void);
-#endif
-
-#endif /* CONFIG_CLFLUSH_DETECT */
-
-
-#if defined(CONFIG_CACHE_LINE_SIZE_DETECT)
-size_t sys_cache_line_size;
-static void init_cache_line_size(void)
-{
-	sys_cache_line_size = z_cache_line_size_get();
-}
-#else
-#define init_cache_line_size() do { } while ((0))
-#endif
-
-static int init_cache(struct device *unused)
-{
-	ARG_UNUSED(unused);
-
-	init_cache_flush();
-	init_cache_line_size();
-
-	return 0;
-}
-
-SYS_INIT(init_cache, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
```

Comment:
I don't think you should remove it, this was flushing the cache during initialization.

Reply:
Yes, this should not be removed. Added this back. Thanks